GeometryCore/src/Polygon.cpp

```diff
@@ -58,13 +72,47 @@ namespace geom_utils
     // Calculate and return convex hull for the polygon
     Polygon Polygon::convexHull() const 
     { 
-        return {}; 
+        if (points.size() <= 3) return *this; //can't bild a polygon from 2 points or less
+        Polygon convexHull;
+        auto pos_of_the_most_right_point = min_element(points.begin(), points.end(), [](auto a, auto b) {return b.x > a.x; }) - points.begin();
```

Comment:
I think there is missing `#inlcude <algorithm>` for `min_element`. Also, `min_element` is located inside `std::` namespace.

Reply:
But i did not have any errors. May be an error did not occur because the <algorithm> is included in Point.h, and Point.h is included in Polygon.h?
Anyway, i will add std:: and <algorithm>

Reply:
> But i did not have any errors. May be an error did not occur because the `<algorithm>` is included in `Point.h`, and `Point.h` is included in `Polygon.h`?

Maybe, thats the case. But never rely that smth is included somewhere, so I dont have to include smth. What if we remove Point.h from Polygon.h or make other changes with includings? This change will trigger other sequence of changes. So, always explicitly include things which you need.

Reply:
OK, i will know it